src/components/JobCard/__test__/JobCard.test.tsx

```diff
@@ -0,0 +1,101 @@
+import { describe, expect, it, vi } from 'vitest'
+import { render } from '@/tests/customRender'
+import { screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import JobCard from '../JobCard'
+
+// Mock data for the job
+const mockJob = {
+    title: 'Software Engineer',
+    company: 'Tech Corp',
+    location: 'Tokyo, Shibuya',
+    salary: '600,000 - 12,000,000',
+    tags: ['React', 'JavaScript', 'Frontend'],
+    logoUrl: 'https://via.placeholder.com/150',
+    publicationDate: '31/01/2025',
+    jobPostingUrl: 'https://jobpostingurl.com',
+}
+
+describe('JobCard Component', () => {
+    it('should render the job details correctly', () => {
+        render(<JobCard job={mockJob} />)
+
+        // Check the job title
+        expect(screen.getByText(mockJob.title)).toBeInTheDocument()
```

Comment:
Preference for `async` functions and `findBy` which are more performant and do retries. This gives us more confidence when tests run in CI/CD. See: https://testing-library.com/docs/dom-testing-library/api-async/#findby-queries.

Also let's use the stronger assertion `toBeVisible()`

```suggestion
        const title = await screen.findByText(mockJob.title)
        expect(title).toBeVisible()
```